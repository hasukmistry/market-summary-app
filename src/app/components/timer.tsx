import React from 'react';

type Props = {
    seconds: number,
};

export const Timer: React.FC<Props> = ({ seconds } : Props ) => {
    return (
        <div className="mr-2" style={ timerStyles }>
            { seconds }
        </div>
    );
};

const timerStyles = {
    padding: '5px 14px',
    backgroundColor: '#dff0d8',
    border: '1px solid #d6e9c6',
    borderRadius: '15px',
    color: '#3c763d',
    fontWeight: '700',
    cursor: 'not-allowed'
}