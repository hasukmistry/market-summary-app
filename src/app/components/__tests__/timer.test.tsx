import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Timer } from '../timer';

const shallowRenderer = createRenderer();

describe('<Timer />', () => {
  it('should match snapshot', () => {
    shallowRenderer.render(<Timer seconds={60}/>);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});