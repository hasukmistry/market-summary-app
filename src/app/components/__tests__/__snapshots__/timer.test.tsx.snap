// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> should match snapshot 1`] = `
<div
  className="mr-2"
  style={
    Object {
      "backgroundColor": "#dff0d8",
      "border": "1px solid #d6e9c6",
      "borderRadius": "15px",
      "color": "#3c763d",
      "cursor": "not-allowed",
      "fontWeight": "700",
      "padding": "5px 14px",
    }
  }
>
  60
</div>
`;
